app/discovery/provider/consulcatalog/list.go

```diff
@@ -0,0 +1,80 @@
+package consulcatalog
+
+import (
+	"fmt"
+	log "github.com/go-pkgz/lgr"
+	"github.com/umputun/reproxy/app/discovery"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+func (cc *ConsulCatalog) List() ([]discovery.URLMapper, error) {
```

Comment:
having some parts of the same ConsulCatalog in two files is confusing. I don't see a goor reason to split it, files are not that large